### ISSUE TITLE
MD013: line length should not be violated for html when `strict = false`

### ISSUE THREAD

**Issue by chrisgrieser** [rvben/rumdl#535]
### rumdl version

rumdl 0.1.53

### Installation method

Pre-built binary

### Bug description

MD013's `strict` config disables line length warnings for URLs, but does not consider html with often also results in lengthy lines you normally do not want to take multiple lines. The following example are badges added to a GitHub readme:

<img width="1286" height="185" alt="Image" src="https://github.com/user-attachments/assets/3f8212c6-2a11-4064-891e-ff49081c8eb8" />

Splitting badges like that up into multiple lines in a way that MD013 is not violated is tricky, it would require unnecessary breaking up like this:

<img width="1175" height="149" alt="Image" src="https://github.com/user-attachments/assets/b433a264-9924-4243-9d93-a7e2054bbd95" />

### Steps to reproduce

```md
# Demo
<a href="https://dotfyle.com/plugins/chrisgrieser/nvim-rulebook"><img alt
="badge" src="https://dotfyle.com/plugins/chrisgrieser/nvim-rulebook/shield"/></a>
```

### Expected behavior

`strict = false` (the default) should not report the HTML above.

### Actual behavior

The line is reported

### Configuration

```toml
# this is already the default though
[line-length]
strict = false
```

### Operating system

macOS

**Comment by rvben** [comment]
Hi @chrisgrieser, thanks for the issue. While markdownlint triggers in the same way, I agree that it's better to ignore it in certain cases when `strict = false` is set. In some cases we might choose to 'be better' than markdownlint and this is a good example. As such this could be considered an 'enhancement' too. Anyway, a fix/implementation for this is upcoming in 0.1.54. :)

**Comment by rvben** [comment]
Fixed in [v0.1.54](https://github.com/rvben/rumdl/releases/tag/v0.1.54).

When `strict = false` (the default), MD013 now exempts lines consisting entirely of inline HTML. This covers patterns like badge links (`<a href="..."><img .../></a>`), standalone `<img>` tags, and other inline HTML elements that can't reasonably be split across lines.

The exemption uses two-tier detection:
1. Lines where all content is inside HTML tags (e.g., `<kbd>Ctrl</kbd><kbd>C</kbd>`)
2. Lines that start and end with tags and contain URL-bearing attributes (`href`, `src`, `srcset`, `poster`)

HTML-only lines are also treated as paragraph boundaries in reflow mode, so they won't be merged into adjacent prose.

Thanks for reporting this, @chrisgrieser!
